File: trip_service/src/main/java/pl/edu/pg/trip/trip/management/dto/AddTripEvent.java
package pl.edu.pg.trip.trip.management.dto;

import lombok.Builder;
import lombok.Data;
import lombok.extern.jackson.Jacksonized;
import pl.edu.pg.trip.enity.Trip;

import java.util.function.Function;

@Data
@Builder
@Jacksonized
public class AddTripEvent {
    private Long hotelId;
    private Long startFlightId;
    private Long endFlightId;

    public static Function<AddTripEvent, Trip> eventToEntityMapper() {
        return event -> Trip.builder()
                .hotelId(event.hotelId)
                .startFlightId(event.startFlightId)
                .endFlightId(event.endFlightId)
                .build();
    }
}

File: tour_operator/trips/AddTripEvent.py
import json


class AddTripEvent:
    def __init__(self, hotel, startFlight, endFlight):
        self.hotelId = hotel.id
        self.startFlightId = startFlight.id
        self.endFlightId = endFlight.id

    def getAddTripMessage(self):
        trip = {
            "hotelId": self.hotelId,
            "startFlightId": self.startFlightId,
            "endFlightId": self.endFlightId
        }
        return json.loads(trip)
